storage/src/test/java/org/apache/kafka/tiered/storage/integration/OffloadAndTxnConsumeFromLeaderTest.java

```diff
@@ -97,7 +94,10 @@ protected void writeTestSpecifications(TieredStorageTestBuilder builder) {
     }
 
     private static RemoteFetchCount getRemoteFetchCount() {
-        FetchCountAndOp segmentFetchCountAndOp = new FetchCountAndOp(6, EQUALS_TO);
+        // Ideally, each remote-log segment should be fetched only once. For 6 segments, we would have 6 fetch-counts.
+        // But, the client can retry the FETCH request, to make the test deterministic, increasing the fetch-count
+        // to be at-max of 12 (2 times of fetch-count).
```

Comment:
This should remove the flakiness if the client retries in slower environments like Jenkins.